scipy/linalg/tests/test_basic.py

```diff
@@ -1334,6 +1334,46 @@ def test_axis_args(self):
         assert_equal(x.shape, (4, 2, 3))
         assert_allclose(x, expected)
 
+class TestBalance(TestCase):
+    
+    def test_string_arg(self):
+        assert_raises(ValueError, balance, 'Some string for fail')
+        
+    def test_infnan_arg(self):
+        assert_raises(ValueError, balance, np.array([[1,2],[3,np.inf]]))
+        assert_raises(ValueError, balance, np.array([[1,2],[3,np.nan]]))        
+        
+    def test_scaling(self):
+        _, y = balance(np.array([[1000, 1], [1000, 0]]))
+        assert_allclose(int(np.diff(np.log2(np.diag(y)))), 5)
```

Comment:
not super difficult, but a comment to spell out why this is right never hurts (``sqrt(1000-1) ~= 2**5``)